### PR TITLE
TDL-6737 added pylint and resolved the errors

### PR DESCRIPTION
**Pull request by namrata270998**
# Description of change

- Added pylint in the circleci and fixed its corresponding errors

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_bing_ads/__init__.py [hunk 1]
```diff
@@ -836,14 +830,12 @@ def get_report_request_id(client, account_id, report_stream, report_name,
...
                          start_date, end_date):
-    LOGGER.info(
-        'Syncing report for account {}: {} - from {} to {}'
-        .format(account_id, report_name, start_date, end_date)
-    )
+    LOGGER.info('Syncing report for account %s: %s - from %s to %s',
+                account_id, report_name, str(start_date), str(end_date))
 
-    report_request = client.factory.create('{}Request'.format(report_name))
+    report_request = client.factory.create(f'{report_name}Request'.format(report_name)) # pylint: disable=consider-using-f-string
```
Comment (dbshah1212) [inline tap_bing_ads/__init__.py:838]: .format is not required if you are using f-stream. Please check the usage of f-stream

File: tap_bing_ads/__init__.py [hunk 2]
```diff
@@ -61,13 +61,12 @@ class InvalidDateRangeEnd(Exception):
...
 def log_service_call(service_method, account_id):
-    def wrapper(*args, **kwargs):
-        log_args = list(map(lambda arg: str(arg).replace('\n', '\\n'), args)) + \
-                   list(map(lambda kv: '{}={}'.format(*kv), kwargs.items()))
-        LOGGER.info('Calling: {}({}) for account: {}'.format(
-            service_method.name,
-            ','.join(log_args),
-            account_id))
+    def wrapper(*args, **kwargs): # pylint: disable=inconsistent-return-statements
+        log_args = list(map(lambda arg: str(arg).replace('\n', '\\n'), args)) + list(map(lambda kv: '{}={}'.format(*kv), kwargs.items())) # pylint: disable=consider-using-f-string
```
Comment (dbshah1212) [inline tap_bing_ads/__init__.py:65]: Use f-string and resolve this.

File: tap_bing_ads/__init__.py [hunk 3]
```diff
@@ -497,7 +496,7 @@ def get_selected_fields(catalog_item, exclude=None):
             selected_fields.append(prop)
 
     if any(invalid_selections):
-        raise Exception("Invalid selections for field(s) - {{ FieldName: [IncompatibleFields] }}:\n{}".format(json.dumps(invalid_selections, indent=4)))
+        raise Exception("Invalid selections for field(s) - {{ FieldName: [IncompatibleFields] }}:\n{}".format(json.dumps(invalid_selections, indent=4))) # pylint: disable=consider-using-f-string
```
[the remark this answers is not in the document]
Comment (namrata270998) [reply tap_bing_ads/__init__.py]: done

File: tap_bing_ads/__init__.py (hunk 2 above)
Comment (namrata270998) [reply tap_bing_ads/__init__.py:65]: not changed this, as it is a bit complex logic

**Review by dbshah1212 — APPROVED** [review]

File: tap_bing_ads/__init__.py [hunk 4]
```diff
@@ -358,7 +357,7 @@ def discover_core_objects():
     return core_object_streams
 
 def get_report_schema(client, report_name):
-    column_obj_name = '{}Column'.format(report_name)
+    column_obj_name = '{}Column'.format(report_name) # pylint: disable=consider-using-f-string
```
Comment (savan-chovatiya) [inline tap_bing_ads/__init__.py:360]: We can disable `consider-using-f-string` directly into the config.yml command and remove all the disabled comments in the code because it is present at multiple places.

Comment (namrata270998) [reply tap_bing_ads/__init__.py:360]: done

File: tap_bing_ads/__init__.py [hunk 5]
```diff
@@ -688,9 +686,7 @@ def stream_report(stream_name, report_name, url, report_time):
         response = SESSION.get(url, headers={'User-Agent': get_user_agent()})
 
     if response.status_code != 200:
-        raise Exception('Non-200 ({}) response downloading report: {}'.format(
-            response.status_code, report_name))
-
+        raise Exception('Non-200 ({}) response downloading report: {}'.format(response.status_code, report_name))
```
Comment (dbshah1212) [inline tap_bing_ads/__init__.py:689]: Reverte this change if it is not throwing any pylint error.

File: tap_bing_ads/__init__.py [hunk 6]
```diff
@@ -864,8 +855,7 @@ def build_report_request(client, account_id, report_stream, report_name,
     report_columns = client.factory.create(
         'ArrayOf{}Column'.format(report_name)
     )
-    getattr(report_columns, '{}Column'.format(report_name)) \
-        .append(selected_fields)
+    getattr(report_columns, '{}Column'.format(report_name)).append(selected_fields)
```
Comment (dbshah1212) [inline tap_bing_ads/__init__.py:858]: revert this change if it is not throwing any pylint error.

Comment (namrata270998) [reply tap_bing_ads/__init__.py]: done

File: tap_bing_ads/__init__.py (hunk 5 above)
Comment (namrata270998) [reply tap_bing_ads/__init__.py]: done

**Review by karanpanchal-crest — APPROVED** [review]

**Review by umangagrawal-crest — APPROVED** [review]

**Review by savan-chovatiya — APPROVED** [review]

File: tap_bing_ads/__init__.py [hunk 7]
```diff
@@ -61,13 +61,12 @@ class InvalidDateRangeEnd(Exception):
...
-                   list(map(lambda kv: '{}={}'.format(*kv), kwargs.items()))
-        LOGGER.info('Calling: {}({}) for account: {}'.format(
-            service_method.name,
-            ','.join(log_args),
-            account_id))
+    def wrapper(*args, **kwargs): # pylint: disable=inconsistent-return-statements
+        log_args = list(map(lambda arg: str(arg).replace('\n', '\\n'), args)) + list(map(lambda kv: '{}={}'.format(*kv), kwargs.items()))
+        LOGGER.info('Calling: %s(%s) for account: %s',
+                    service_method.name,
+                    ','.join(log_args),
```
Comment (KrisPersonal) [inline tap_bing_ads/__init__.py:68]: What is the difference between what was before and the changes done?

Comment (namrata270998) [reply tap_bing_ads/__init__.py:68]: We have removed `format` and used `%s` instead, as pylint was throwing error for `format`

**Review by kspeer825 — APPROVED** [review]
